File: KoohiiAite/src/main/java/com/diarmaidlindsay/koohii/model/SampleWord.java
package com.diarmaidlindsay.koohii.model;

/**
 * Represent an entry in the sample_word table
 */
public class SampleWord {
    int id;
    int heisigId;
    String kanjiWord;
    String hiraganaReading;
    String englishMeaning;
    String category;
    int frequency;

    public SampleWord(int id, int heisigId, String kanjiWord, String hiraganaReading, String englishMeaning, String category, int frequency) {
        this.id = id;
        this.heisigId = heisigId;
        this.kanjiWord = kanjiWord;
        this.hiraganaReading = hiraganaReading;
        this.englishMeaning = englishMeaning;
        this.category = category;
        this.frequency = frequency;
    }

    public int getId() {
        return id;
    }

    public int getHeisigId() {
        return heisigId;
    }

    public String getKanjiWord() {
        return kanjiWord;
    }

    public String getHiraganaReading() {
        return hiraganaReading;
    }

    public String getEnglishMeaning() {
        return englishMeaning;
    }

    public String getCategory() {
        return category;
    }

    public int getFrequency() {
        return frequency;
    }
}

File: app/src/main/java/tech/diarmaid/koohiiaite/database/entity/SampleWord.kt
package tech.diarmaid.koohiiaite.database.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

/**
 * Represent an entry in the sample_word table
 */
@Entity(tableName = "sample_words",
        foreignKeys = [ForeignKey(entity = HeisigKanji::class, parentColumns = ["id"], childColumns = ["heisig_id"])])
data class SampleWord(@PrimaryKey val id: Int,
                      @ColumnInfo(name = "heisig_id") val heisigId: Int,
                      @ColumnInfo(name = "kanji_word") val kanjiWord: String,
                      @ColumnInfo(name = "hiragana_reading") val hiraganaReading: String,
                      @ColumnInfo(name = "english_meaning") val englishMeaning: String,
                      val category: String?,
                      val frequency: Int)
